lib/much-result.rb

```diff
@@ -197,7 +197,8 @@ def inspect
     "#<#{self.class}:#{"0x0%x" % (object_id << 1)} "\
       "#{success? ? "SUCCESS" : "FAILURE"} "\
       "#{"@description=#{@description.inspect} " if @description}"\
-      "@sub_results=#{@sub_results.inspect}>"
+      "@sub_results=#{@sub_results.inspect} "\
+      "Keys: #{@data.to_h.keys.reject { |key| key.to_s.start_with?("much_result_") }}>"
```

Comment:
@pdobb what do you think about formalizing this concept as "attributes" a bit? I also like calling it "attribute_names" better than the implementation-driven "Keys". e.g.:

```ruby
# or whatever better way to filter down the @data openstruct key/values
def attributes
  attribute_names.reduce({}) { |acc, attribute_name|
    acc[attribute_name] = @data[attribute_name]
  }
end

def attribute_names
  @data.to_h.keys.reject { |key| key.to_s.start_with?("much_result_")
end
```

Then you can just do:

```suggestion
      "@sub_results=#{@sub_results.inspect} "\
      "attribute_names: #{attribute_names}>"
```

Reply:
Looks great. I'll get back to this when I can devote a bit more time to learning this "Assert" gem for testing. Or feel free to finish it out. Thanks